test/AccountSystem.t.sol

```diff
@@ -0,0 +1,1718 @@
+
+        vm.startPrank(player1);
+
+        vm.expectEmit(address(accountSystem));
+        emit AccountSystem.AccountCreated(
+            expectedAccountAddress,
+            player1,
+            AccountVersion
+        );
+        (address accountAddress, bool created) = accountSystem.createAccount(
+            player1
+        );
+
+        vm.stopPrank();
+
+        vm.assertEq(expectedAccountAddress, accountAddress);
+        vm.assertTrue(created);
+
+        address actualPlayer = accountSystem.accounts(player1).player();
+        vm.assertEq(actualPlayer, player1);
+    }
+
+    function test_account_creation_is_idempotent() public {
+        address expectedAccountAddress = accountSystem.calculateAccountAddress(
+            player1
+        );
+
+        address actualAccountAddress = address(accountSystem.accounts(player1));
+        vm.assertEq(actualAccountAddress, address(0));
+
+        vm.startPrank(player1);
+
+        (address accountAddress, bool created) = accountSystem.createAccount(
+            player1
+        );
+
+        vm.assertEq(expectedAccountAddress, accountAddress);
+        vm.assertTrue(created);
+
+        address actualPlayer = accountSystem.accounts(player1).player();
+        vm.assertEq(actualPlayer, player1);
+
+        (accountAddress, created) = accountSystem.createAccount(player1);
+
+        vm.assertEq(expectedAccountAddress, accountAddress);
+        vm.assertEq(created, false);
+
+        vm.stopPrank();
+    }
+
+    function test_account_creation_is_permissionless() public {
+        address expectedAccountAddress = accountSystem.calculateAccountAddress(
+            player1
+        );
+
+        address actualAccountAddress = address(accountSystem.accounts(player1));
+        vm.assertEq(actualAccountAddress, address(0));
+
+        // Even player2 can create an account for player1.
+        vm.startPrank(player2);
+
+        vm.expectEmit(address(accountSystem));
+        emit AccountSystem.AccountCreated(
+            expectedAccountAddress,
+            player1,
+            AccountVersion
+        );
+        accountSystem.createAccount(player1);
+
+        vm.stopPrank();
+
+        actualAccountAddress = address(accountSystem.accounts(player1));
+        vm.assertEq(expectedAccountAddress, actualAccountAddress);
+
+        address actualPlayer = accountSystem.accounts(player1).player();
+        vm.assertEq(actualPlayer, player1);
+    }
+
+    function test_withdraw_native_token() public {
+        vm.startPrank(player1);
+        (address accountAddress, ) = accountSystem.createAccount(player1);
+        DegenCasinoAccount account = accountSystem.accounts(player1);
+
+        uint256 initialPlayerBalance = player1.balance;
+
+        vm.assertEq(accountAddress.balance, 0);
+
+        uint256 depositAmount = startingBalance / 10;
+        uint256 withdrawalAmount = depositAmount / 2;
+
+        accountAddress.call{value: depositAmount}("");
```

Comment:
This and lines
179, 290, 315, 410, 482, 544, 594
Share warning (9302) where the low level call is return values are not used. 
Maybe use the bool
vm.assertEq(success, true);